nixos/modules/services/web-apps/mastodon.nix

```
@@ -190,12 +220,53 @@ in {
         type = lib.types.port;
         default = 55002;
       };
+
       sidekiqThreads = lib.mkOption {
-        description = lib.mdDoc "Worker threads used by the mastodon-sidekiq service.";
+        description = lib.mdDoc "Worker threads used by the mastodon-sidekiq-all service. Ignored if `sidekiqProcesses` is configured.";
         type = lib.types.int;
         default = 25;
       };
 
+      sidekiqProcesses = lib.mkOption {
+        description = "How many Sidekiq processes should be used to handle background jobs, and which job classes they handle.";
+        type = with lib.types; attrsOf (submodule {
+          options = {
+            jobClasses = lib.mkOption {
+              type = listOf (enum [ "default" "push" "pull" "mailers" "scheduler" "ingress" ]);
+              description = "If not empty, which job classes should be executed by this process. *Only one process should handle the 'scheduler' class.*";
+            };
+            threads = lib.mkOption {
+              type = int;
+              description = "Number of threads this process should use for executing jobs.";
+            };
+          };
+        });
+        default = {
+          all = {
+            jobClasses = [ ];
+            threads = cfg.sidekiqThreads;
```

```suggestion
            threads = 25;
```

https://github.com/NixOS/nixpkgs/actions/runs/3526803585/jobs/5915148238 failing because:
> Usually this means that an option attribute that ends up in documentation (eg `default` or `description`) depends on the restricted module arguments `config` or `pkgs`.

---

thanks for taking a look, I made `threads` accept a null value which causes fallback to `cfg.sidekiqThreads`, and then made the default `cfg.sidekiqProcesses` specify null.